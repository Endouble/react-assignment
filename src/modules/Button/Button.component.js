import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { colors } from '../../styles/vars';
import { boxShadow } from '../../styles/mixins';

const Button = props => (
    <StyledButton {...props}>
        <button className="pokeButton" type="button">
            {props.copy}
        </button>
    </StyledButton>
);

/*
    Button Styles
*/
const StyledButton = styled.div`
    .pokeButton {
        width: ${props => props.sizeIndex * 10}px;
        height: ${props => props.sizeIndex * 10}px;
        left: 0;
        top: 0;
        border-radius: 50%;
        position: relative;
        cursor: pointer;
        transition: all 250ms ease;
        ${props => (!props.isPlain && boxShadow(8))};
        border: ${props => (props.isPlain ? '4px solid' : 'none')};
        background: ${props => props.backColor || colors.gray};

        &:hover {
            left: ${props => (!props.isPlain && '8px' : '0')};
            top: ${props => (!props.isPlain && '8px' : '0')};
            ${props => (!props.isPlain && boxShadow(0))};

        }

        &:before {
            content: '';
            position: absolute;
            background: white;
            width: 10px;
            height: 10px;
            border-radius: 50%;
            left: 20%;
            top: 20%;
        }

        &:after {
            content: '';
            position: absolute;
            width: 100%;
            height: 100%;
            border-radius: 50%;
            box-shadow: 5px 5px 0 0 black;
            opacity: 0.2;
            bottom: 10%;
            right: 10%;
        }
    }

`;

/*
    Button propTypes
*/
Button.propTypes = {
    copy: PropTypes.string,
    className: PropTypes.string,
    backColor: PropTypes.string,
    isPlain: PropTypes.bool,
    sizeIndex: PropTypes.number,
};

Button.defaultProps = {
    copy: '',
    className: '',
    backColor: '',
    isPlain: true,
    sizeIndex: 5,
};

export default Button;
