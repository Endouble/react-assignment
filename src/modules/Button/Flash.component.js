import React from 'react';
import styled from 'styled-components';

import Button from './';

import { boxShadow } from '../../styles/mixins';
import { colors } from '../../styles/vars';

const Flash = () => (
    <StyledFlash>
        <Button isPlain backColor={colors.blue} />
    </StyledFlash>
);

/*
    Flash Styles
*/
const StyledFlash = styled.div`
    ${boxShadow(5)}
    background: white;
    border-radius: 50%;
    padding: 30px;
    display: inline-block;
    position: relative;

    &:after {
        content: '';
        left: 0px;
        right: 0;
        bottom: 0;
        top: 0;
        background: transparent;
        display: block;
        position: absolute;
        border-radius: 50%;
        border: 10px solid;
        border-color: transparent ${colors.blue} ${colors.blue} transparent;
        transform: rotateZ(45deg);
    }
`;

export default Flash;
