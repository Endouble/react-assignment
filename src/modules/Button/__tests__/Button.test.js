import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from '../';

configure({ adapter: new Adapter() });

describe('Button', () => {
    it('renders correctly', () => {
        const component = shallow(<Button />);
        expect(component).toMatchSnapshot();
    });
    it('renders not plain Button', () => {
        const component = shallow(<Button isPlain={false} />);
        const type = component.find('.pokeButton').type();
        expect(type).toEqual('button');
    });
});
