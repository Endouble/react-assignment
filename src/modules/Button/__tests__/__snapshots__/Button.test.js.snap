// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<styled.div
  backColor=""
  className=""
  isPlain={true}
  sizeIndex={5}
>
  <div
    className="pokeButton"
  />
</styled.div>
`;
